### PR TITLE
Server Reports extension: fix for start and stop xevent sessions

### PR DESCRIPTION
**Pull request by dzsquared**
Before: "Server Reports" extension v0.2.2, TempDB tab is unable to start/stop XEvent sessions.  Task triggers query files to open into file windows instead of query windows, leading to some odd behaviors. (user is unable to close editor without closing application)

serverreports-broken

After: TempDB tab start/stop XEvent sessions tasks are run in the background with a queryprovider.
serverreports-background

image

Changes:
- queries startEvent.sql and stopEvent.sql return a query result indicating success/error from try/catch
- tempdb tab tasks utilize a query provider to execute the queries instead of requiring further user interaction
- contention help link utilizes vscode.open instead of 'openurl' module to provide user with open/copy/cancel dialog

### CODE REVIEW

File: samples/serverReports/src/controllers/mainController.ts [hunk 1]
```diff
@@ -38,21 +37,28 @@ export default class MainController extends ControllerBase {
...
+		if (seResult.rowCount > 0 && seResult.rows[0][0].displayValue === '0') {
+			vscode.window.showInformationMessage(seResult.rows[0][1].displayValue);
+			let setRefreshState = ( fileName === 'startEvent.sql' ) ? true : false;
+			vscode.commands.executeCommand('azdata.widget.setAutoRefreshState', 'type-of-contention', connection.id, setRefreshState);
+			vscode.commands.executeCommand('azdata.widget.setAutoRefreshState', 'metadata-contention', connection.id, setRefreshState);
+			vscode.commands.executeCommand('azdata.widget.setAutoRefreshState', 'allocation-contention', connection.id, setRefreshState);
+		} else if (seResult.rowCount > 0 && seResult.rows[0][0].displayValue === '1') {
+			vscode.window.showErrorMessage(seResult.rows[0][1].displayValue);
+		} else {
+			vscode.window.showErrorMessage('Something has gone wrong, better error msg here.');
```
Comment (Charles-Gagnon) [inline samples/serverReports/src/controllers/mainController.ts:60]: Would suggest actually including a better error message here =) And this should be localized

File: samples/serverReports/src/sql/startEvent.sql [hunk 2]
```diff
@@ -24,9 +22,11 @@ IF NOT EXISTS (SELECT * FROM sys.dm_xe_sessions  WHERE name = 'ObjectContention'
         ALTER EVENT SESSION [ObjectContention] ON SERVER
         STATE =  START
     END
+SELECT 0 AS RESULTCODE, 'XEvent sessions started for PageContention and ObjectContention' AS RESULTMSG
```
Comment (Charles-Gagnon) [inline samples/serverReports/src/sql/startEvent.sql:25]: It's kind of weird to have the success messages in the SQL script - why not have them in the extension so they can be properly localized?

**Review by Charles-Gagnon — APPROVED** [review]

**Review by kburtram — APPROVED** [review]

**Comment by kburtram** [comment]
@dzsquared please merge when you like.  Let me know if you need any help publishing the update.

**Comment by dzsquared** [comment]
> @dzsquared please merge when you like. Let me know if you need any help publishing the update.

sorry - I haven't had a chance to make the error messages localizable - should be within the next week.

File: samples/serverReports/src/sql/startEvent.sql (hunk 2 above)
Comment (dzsquared) [reply samples/serverReports/src/sql/startEvent.sql]: @Charles-Gagnon I took a whack at setting the messages up for localization from the actions, but I skipped localizing the package.json for now. thanks for the feedback!

**Pull request merged**